### PR TITLE
feat: invoke inside workspace interceptor

### PR DESCRIPTION
**Pull request by f0rmiga**
This is a follow-up from https://github.com/aspect-build/aspect-cli/pull/79 with includes fixes and the actual usage of the functionality.

### CODE REVIEW

File: cmd/aspect/clean/clean.go [hunk 1]
```diff
@@ -53,16 +61,23 @@ Workaround inconistent state:
 	Such problems are fixable and these bugs are a high priority.
 	If you ever find an incorrect incremental build, please file a bug report,
 	and only use clean as a temporary workaround.`,
-		RunE: b.Run,
+		RunE: pathutils.InvokeCmdInsideWorkspace(func(workspaceRoot string, cmd *cobra.Command, args []string) (exitErr error) {
+			bzl.SetWorkspaceRoot(workspaceRoot)
+			isInteractive := isatty.IsTerminal(os.Stdout.Fd()) || isatty.IsCygwinTerminal(os.Stdout.Fd())
```
Comment (alexeagle) [inline cmd/aspect/clean/clean.go:66]: didn't we fix this already? we should honor the `--interactive` CLI flag too

File: pkg/pathutils/pathutils.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	wd string,
+	toRun RunWorkspaceFn,
+) CobraRunEFn {
+	return func(cmd *cobra.Command, args []string) (exitErr error) {
+		workspacePath, err := finder.Find(wd)
+		if err != nil {
+			return fmt.Errorf("failed to run command %q: %w", cmd.Use, err)
+		}
+		if workspacePath == "" {
+			err = fmt.Errorf("the current working directory %q is not a Bazel workspace", wd)
```
Comment (alexeagle) [inline pkg/pathutils/pathutils.go:49]: note for later (maybe a TODO?) we should offer to help if the user actually meant the cwd to be a workspace - go interactive and into a setup mode

**Review by alexeagle — APPROVED** [review]

File: cmd/aspect/clean/clean.go (hunk 1 above)
Comment (f0rmiga) [reply cmd/aspect/clean/clean.go:66]: I guess there are some leftovers. I'll do a second pass on another PR to fix them all.

File: pkg/pathutils/pathutils.go (hunk 2 above)
Comment (f0rmiga) [reply pkg/pathutils/pathutils.go:49]: Created https://github.com/aspect-build/aspect-cli/issues/85.

**Pull request merged**